### ISSUE TITLE
Community Edition Features

### ISSUE THREAD

**Issue by alxxyz** [arangodb/arangodb#16136]
Where I can find the list with all *Community Edition Features* ?

There is no link on the page https://www.arangodb.com/subscriptions

**Comment by Simran-B** [comment]
For what it is worth, the Community Edition has all features except those exclusive to the Enterprise Edition.
Are you looking for a detailed list of all features?

**Comment by alxxyz** [comment]
It will be great to have a detailed list, as we are now deciding which database to use.
I've found this list https://www.arangodb.com/features-december-2018/ , but it is from 2018, and I don't know if it is actual.

**Comment by Simran-B** [comment]
A few things have changed and some newer features aren't listed, but the comparison is otherwise still applicable.

- **Data Storage**: ~mostly memory~ --> data on disk, including indexes, with a hot set in memory
- **Storage Engines**: ~RocksDB & MMfiles~ --> only RocksDB since v3.7.0
- **Sync and Async Replication**: The Leader/Follower deployment mode with async replication is deprecated since v3.9.0, but Active Failover using multiple single servers with async replication is still supported. They are all Community Edition features.
- **Cluster deployments** with sync replication are fully supported in Community Edition
- **Datacenter to Datacenter Replication** (DC2DC) is only available in the Enterprise Edition

We will add updated feature lists to the documentation soon.

**Comment by Delicious-Bacon** [comment]
> * **Sync and Async Replication**: The Leader/Follower deployment mode with async replication is deprecated since v3.9.0, but Active Failover using multiple single servers with async replication is still supported. They are all Community Edition features.
> * **Cluster deployments** with sync replication are fully supported in Community Edition
> * **Datacenter to Datacenter Replication** (DC2DC) is only available in the Enterprise Edition
> 
> We will add updated feature lists to the documentation soon.

Is there an updated documentation somewhere?

Also, how do you define a "cluster" and a "datacenter"? Is "cluster" multiple ArangoDB servers running on a single physical machine, or ArangoDB servers running on multiple machines?

**Comment by Simran-B** [comment]
We added a feature overview as well as detailed feature lists for the Community and Enterprise Edition to the 3.10 documentation:
https://www.arangodb.com/docs/devel/features.html

"Cluster" is an ArangoDB deployment mode that is comprised of an Agency, Coordinators, and DB-Servers:
https://www.arangodb.com/docs/devel/architecture-deployment-modes-cluster-architecture.html#structure-of-an-arangodb-cluster
Each Agent, Coordinator, and DB-Server node is an _arangod_ process (ArangoDB server executable started with different options). Whether you run them on the same or different machines is up to you, but running them all on one machine has obvious drawbacks regarding resilience and maintenance. A common setup is to use three machines, and run an Agent, a Coordinator, and a DB-Server per machine. The machines should be connected to the same local network (e.g. one datacenter) because of the synchronous replication. You cannot geo-distribute the nodes of a cluster.

DC2DC replication is asynchronous and a somewhat high network latency between the datacenters is expected, as they should be far apart geographically because this deployment mode is for disaster recovery. The concept is to have a primary cluster that is replicated to one or more secondary clusters that are topologically the same as the primary cluster and that you can't modify data in.
https://www.arangodb.com/docs/devel/architecture-deployment-modes-dc2-dc-introduction.html

**Comment by Delicious-Bacon** [comment]
Those links are helpful, thank you!

From the docs:
> DB-Servers are the ones where the data is actually hosted. They host shards of data and using synchronous replication a DB-Server may either be leader or follower for a shard.

Could you explain how would the DB Server function in a Community edition cluster?

For example, let's say I have 3 machines hosting their DB Server each. Those 3 DB Servers are part of a cluster. Which way would the cluster work, considering the "sharding" the docs mention.

```
// Servers partition data between each other
S1 - Shard 1
S2 - Shard 2
S3 - Shard 3
```
OR
```
// Each server has all the data
S1 - Shard 1, Shard 2, Shard 3
S2 - Shard 1, Shard 2, Shard 3
S3 - Shard 1, Shard 2, Shard 3
```

I've noticed the label that "One shard" feature is available only to the Enterprise edition, but what if I wanted to just have _copies_ of Master data on each Follower?

**Comment by Simran-B** [comment]
The number of shards defines into how many pieces the data will be split, and each shard will only be on a single DB-Server by default. If you set the number of shards to 3, then that would lead to your first example.

If you want copies of the shards on other DB-Servers for resilience, then you need to set a replication factor higher than 1. This is independent of the number of shards. A replication factor of 3 would lead to the second example, where each shard exists three times.

The OneShard feature is for forcing the collections of a database to number of shards = 1, and have all these single shards co-located on a single DB-Server, so that queries can be executed fully locally on that one node. This is mainly of interest for multi tenant setups, where each tenant has its own database.

**Comment by Delicious-Bacon** [comment]
@Simran-B Could you explain the caching in the cluster, please?

The [docs mention](https://www.arangodb.com/docs/devel/aql/execution-and-performance-query-cache.html):
> The AQL query results cache is only available for single servers, i.e. servers that are not part of a cluster setup.

But why is it not possible to cache the AQL query results in a cluster setup? Is it because of sharding or coordinator limitations?

What if I had `min-replication-factor` and `max-replication-factor` of 1? Would that be something like OneShard, and result in:

```
// Each server has all the data, acting as replicas
S1 - Shard 1
S2 - Shard 1
S3 - Shard 1
```

Would that affect the ability to cache AQL query results?

In other words, is it possible to have replicas of 1 shard (no sharding), and would that be able to cache AQL query results?

Can OneShard cache the AQL query results?

I ask about caching because my queries run slow without the cache, so cluster setup would have slow queries.
